### ISSUE TITLE
[STORM-694] Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo

### ISSUE THREAD

**Issue by jira-importer** [apache/storm#4535]
<p>When compiling storm I get an exception in supervisor.clj</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Exception in thread "main" java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo, compiling:(supervisor.clj:16:1)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3558)
	at clojure.lang.Compiler.compile1(Compiler.java:7226)
	at clojure.lang.Compiler.compile1(Compiler.java:7216)
	at clojure.lang.Compiler.compile(Compiler.java:7292)
	at clojure.lang.RT.compile(RT.java:398)
	at clojure.lang.RT.load(RT.java:438)
	at clojure.lang.RT.load(RT.java:411)
	at clojure.core$load$fn__5066.invoke(core.clj:5641)
	at clojure.core$load.doInvoke(core.clj:5640)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invoke(core.clj:5446)
	at clojure.core$compile$fn__5071.invoke(core.clj:5652)
	at clojure.core$compile.invoke(core.clj:5651)
	at clojure.lang.Var.invoke(Var.java:379)
	at clojure.lang.Compile.main(Compile.java:81)
</pre>
</div></div>

<p>After adding an explicit <tt>:require</tt> in <tt>supervisor.clj</tt> the issue seems to be  resolved :</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>diff --git a/storm-core/src/clj/backtype/storm/daemon/supervisor.clj b/storm-core/src/clj/backtype/storm/daemon/supervisor.clj
index b9f9632..5fc5b7c 100644
--- a/storm-core/src/clj/backtype/storm/daemon/supervisor.clj
+++ b/storm-core/src/clj/backtype/storm/daemon/supervisor.clj
@@ -15,6 +15,7 @@
 ;; limitations under the License.
 (ns backtype.storm.daemon.supervisor
   (:import [java.io OutputStreamWriter BufferedWriter IOException])
+  (:require [backtype.storm.daemon.common :as common])
   (:import [backtype.storm.scheduler ISupervisor]
    [backtype.storm.utils LocalState Time Utils]
    [backtype.storm.daemon Shutdownable]
</pre>
</div></div>

---
<details><summary><i>Originally reported by <a title="Duke Bartholomew" href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=camme">camme</a>, imported from: <a href="https://issues.apache.org/jira/browse/STORM-694" target="_blank">Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2015-06-01T21:01:46+01:00
<li><b>imported</b>: 2025-01-24
</ul></i>
</details>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>GitHub user duke-bartholomew opened a pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449</a></p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-694" title="Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo" class="issue-link" data-issue-key="STORM-694"><del>STORM-694</del></a></p>

<p>    Exception in thread "main" java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo, compiling:(supervisor.clj:16:1)</p>


<p>You can merge this pull request into a Git repository by running:</p>

<p>    $ git pull <a href="https://github.com/duke-bartholomew/storm" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/duke-bartholomew/storm</a> master</p>

<p>Alternatively you can review and apply these changes as the patch at:</p>

<p>    <a href="https://github.com/apache/storm/pull/449.patch" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449.patch</a></p>

<p>To close this pull request, make a commit to your master/trunk branch<br/>
with (at least) the following in the commit message:</p>

<p>    This closes #449</p>

<hr />
<p>commit c2b23cfb2fd167aa9d53e16cfa49e7665324d46f<br/>
Author: Bartholomew <duke.bartholomew@gmail.com><br/>
Date:   2015-02-27T09:45:49Z</p>

<p>    add explicit :require</p>

<p>commit 905c14e10b878194b12646240284cbf014d8eadd<br/>
Author: Duke Bartholomew <duke.bartholomew@gmail.com><br/>
Date:   2015-02-27T09:48:45Z</p>

<p>    Merge pull request #1 from duke-bartholomew/<a href="https://issues.apache.org/jira/browse/STORM-694" title="Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo" class="issue-link" data-issue-key="STORM-694"><del>STORM-694</del></a></p>

<p>    add explicit :require</p>

<hr />

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user caofangkun commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-85329513" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-85329513</a></p>

<p>    @duke-bartholomew <br/>
    Does this still a issue? <br/>
    See  SupervisorInfo has already be imported at : <br/>
    <a href="https://github.com/apache/storm/blob/master/storm-core/src/clj/backtype/storm/daemon/supervisor.clj#L21" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/blob/master/storm-core/src/clj/backtype/storm/daemon/supervisor.clj#L21</a>  </p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user duke-bartholomew commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-85680330" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-85680330</a></p>

<p>    @caofangkun <br/>
    I just tried again on the latest master branch, and sadly enough, yes the issue is still there.<br/>
    When executing 'mvn package' or 'mvn test' I still get the same error : <br/>
    ```<br/>
    Exception in thread "main" java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo, compiling:(supervisor.clj:16:1)<br/>
    	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3558)<br/>
    	at clojure.lang.Compiler.compile1(Compiler.java:7226)<br/>
    	at clojure.lang.Compiler.compile1(Compiler.java:7216)<br/>
    	at clojure.lang.Compiler.compile(Compiler.java:7292)<br/>
    	at clojure.lang.RT.compile(RT.java:398)<br/>
    	at clojure.lang.RT.load(RT.java:438)<br/>
    	at clojure.lang.RT.load(RT.java:411)<br/>
    	at clojure.core$load$fn__5066.invoke(core.clj:5641)<br/>
    	at clojure.core$load.doInvoke(core.clj:5640)<br/>
    	at clojure.lang.RestFn.invoke(RestFn.java:408)<br/>
    	at clojure.core$load_one.invoke(core.clj:5446)<br/>
    	at clojure.core$compile$fn__5071.invoke(core.clj:5652)<br/>
    	at clojure.core$compile.invoke(core.clj:5651)<br/>
    	at clojure.lang.Var.invoke(Var.java:379)<br/>
    	at clojure.lang.Compile.main(Compile.java:<br/>
    ```</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user caofangkun commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-85783849" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-85783849</a></p>

<p>    @duke-bartholomew <br/>
    It works very well  when I try to reproduce this issue:<br/>
    ```<br/>
       git clone <a href="https://github.com/apache/storm.git" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm.git</a>  <br/>
       cd storm <br/>
       mvn package -Dmaven.test.skip=true <br/>
    ```</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user duke-bartholomew commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-85884589" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-85884589</a></p>

<p>    @caofangkun <br/>
    indeed, you are right.  If I start from a new clone, it seems to work.<br/>
    Thanks for your reply!</p>

<p>    It still confuses me though why this is not working when I pull <em>master</em> on my already cloned storm repo.<br/>
    But that's most probably a problem on my side then.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user knusbaum commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-87743965" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-87743965</a></p>

<p>    You're including `common` twice, once on line 29, and once on line 18.</p>

<p>    The real solution is to remove the import of `<span class="error">&#91;backtype.storm.daemon.common SupervisorInfo&#93;</span>` and use the record constructor `(->SupervisorInfo ...)` rather than the Java constructor `(SupervisorInfo. ...)` on line 504.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>GitHub user knusbaum opened a pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/492" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492</a></p>

<p>    <a href="https://issues.apache.org/jira/browse/STORM-694" title="Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo" class="issue-link" data-issue-key="STORM-694"><del>STORM-694</del></a>: java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo</p>



<p>You can merge this pull request into a Git repository by running:</p>

<p>    $ git pull <a href="https://github.com/knusbaum/incubator-storm" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/knusbaum/incubator-storm</a> <a href="https://issues.apache.org/jira/browse/STORM-694" title="Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo" class="issue-link" data-issue-key="STORM-694"><del>STORM-694</del></a></p>

<p>Alternatively you can review and apply these changes as the patch at:</p>

<p>    <a href="https://github.com/apache/storm/pull/492.patch" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492.patch</a></p>

<p>To close this pull request, make a commit to your master/trunk branch<br/>
with (at least) the following in the commit message:</p>

<p>    This closes #492</p>

<hr />
<p>commit 0af1d08c690b1bed68669e4fd291a7c186ea185c<br/>
Author: Kyle Nusbaum <kylejnusbaum@gmail.com><br/>
Date:   2015-03-30T17:24:13Z</p>

<p>    Fix for <a href="https://issues.apache.org/jira/browse/STORM-694" title="Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo" class="issue-link" data-issue-key="STORM-694"><del>STORM-694</del></a></p>

<hr />

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user knusbaum commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/492#issuecomment-87761632" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492#issuecomment-87761632</a></p>

<p>    #449 </p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user duke-bartholomew commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-87809406" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-87809406</a></p>

<p>    @knusbaum Thanks for the input :+1: </p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user ptgoetz commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/492#issuecomment-90661810" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492#issuecomment-90661810</a></p>

<p>    +1</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user harshach commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/492#issuecomment-90666963" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492#issuecomment-90666963</a></p>

<p>    +1</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user asfgit closed the pull request at:</p>

<p>    <a href="https://github.com/apache/storm/pull/492" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492</a></p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user harshach commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/492#issuecomment-91033826" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/492#issuecomment-91033826</a></p>

<p>    Thanks @knusbaum  merged it into master.</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user d2r commented on the pull request:</p>

<p>    <a href="https://github.com/apache/storm/pull/449#issuecomment-146644018" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449#issuecomment-146644018</a></p>

<p>    @duke-bartholomew , Seems <a href="https://issues.apache.org/jira/browse/STORM-694" title="Compiling storm 0.10.0 results in java.lang.ClassNotFoundException: backtype.storm.daemon.common.SupervisorInfo" class="issue-link" data-issue-key="STORM-694"><del>STORM-694</del></a> was closed because #492 was merged in.  Should this pull request be closed?</p>

**Comment by jira-importer** [comment]
<i><a href="https://issues.apache.org/jira/secure/ViewProfile.jspa?name=githubbot">githubbot</a>:</i>
<p>Github user asfgit closed the pull request at:</p>

<p>    <a href="https://github.com/apache/storm/pull/449" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/apache/storm/pull/449</a></p>
